llvm/test/CodeGen/NVPTX/cmpxchg.ll

```diff
@@ -0,0 +1,37 @@
+; RUN: llc < %s -march=nvptx64 -mcpu=sm_32 | FileCheck %s --check-prefixes=SM30,CHECK
```

Comment:
This test should be suitable for automatic check generation -- we probably do want to see the details of what we're doing when we do 32-bit CAS on 16-bit data.

Reply:
Do we document how to do that somewhere? I didn't knew that we have automatic check generation!

Reply:
https://llvm.org/docs/TestingGuide.html#generating-assertions-in-regression-tests

Reply:
Seems, that noone autogenerated tests for ptx before, because there wasn't nvptx march